plugins/VstEffect/VstEffectControlDialog.cpp

```diff
@@ -47,7 +47,8 @@ VstEffectControlDialog::VstEffectControlDialog( VstEffectControls * _ctl ) :
 	m_pluginWidget( NULL ),
 
 	m_plugin( NULL ),
-	tbLabel( NULL )
+	tbLabel( NULL ),
+	m_needsEmbed( false )
```

Comment:
This appears to be equivalent to `m_plugin && m_plugin->embetMethod() != "none"`. If so, can we remove the variable and make it a method instead in order to reduce redundancy?

Reply:
We can't, because `VstEffectControlDialog::showEvent` may modify the value? I guess it would be safe to replace `embed_vst` with this variable if `showEvent` is never called while construction.

Reply:
`m_needsEmbed` is a flag to indicate that the VST GUI needs to be embedded but hasn't been yet. The VST GUI seems to be deleted on Qt4 with X11 embedding if the first thing it's added into isn't currently part of the main window hierarchy, so I deferred the embedding to the first time `showEvent` is called, when the `VstEffectControlDialog` will be safely inside its subwindow. This could be replaced by `m_plugin && m_plugin->embetMethod() != "none"`, but the GUI would be re-added to the dialog every time `showEvent` is called.